pkg/trace/agent/tags.go

```diff
@@ -280,14 +284,16 @@ func NormalizeTag(tag string) string {
 			continue
 		}
 
-		c = unicode.ToLower(c)
+		if unicode.IsUpper(c) {
+			utf8.EncodeRune(norm[i:], unicode.ToLower(c))
```

Comment:
Does utf-8 guarantee that `utf8.RuneLen(c) >= utf8.RuneLen(unicode.ToLower(c))`? If not, we might end up with some weird edge case where the last character in a tag needs 1 more byte than what `norm` has and `EncodeRune` assumes `norm[i:]` will contain enough space for `lower(c)`.

Reply:
Yeah, good question. I was pondering the same for the switch cases below, where we measure `c` as uppercase (when it's uppercase) but have written it as lower case. I'll look into this and report back here.

Reply:
To my understanding, they all seem to be the same width. All mappings are listed in the standard library, take a look: https://github.com/golang/go/blob/master/src/unicode/tables.go#L6669. 

References:
[1] https://github.com/golang/go/blob/master/src/unicode/letter.go#L47
[2] https://github.com/golang/go/blob/master/src/unicode/letter.go#L208

Reply:
This case for instance seems to break the assumption: https://github.com/golang/go/blob/master/src/unicode/tables.go#L6687

http://www.fileformat.info/info/unicode/char/17f/index.htm

Upper-case code: 83
Lower-case code: 383 (2 bytes)

Reply:
You're right: https://play.golang.org/p/-weBidY1oza. This would complicate things a bit. I'm wondering if it's worth covering for these edge cases...

Reply:
One idea is to simply not lower-case these exceptions - I think that'd be a valid compromise. What do you think?

Reply:
I added another commit covering this exception. Would be interesting to hear your thoughts if this could be a problem somewhere.